packages/math-extras/src/pointExtras.ts

```diff
@@ -0,0 +1,419 @@
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
+        }
+        outPoint.x = this.x + other.x;
+        outPoint.y = this.y + other.y;
+
+        return outPoint;
+    },
+
+    /**
+     * Subtracts `other` from `this` point and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method subtract
+     * @memberof PIXI.Point#
+     * @param {IPointData} other - The point to subtract to `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the result of the subtraction.
+     */
+    /**
+     * Subtracts `other` from `this` point and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method subtract
+     * @memberof PIXI.ObservablePoint#
+     * @param {IPointData} other - The point to subtract to `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the result of the subtraction.
+     */
+    subtract<T extends IPointData>(other: IPointData, outPoint?: T): T
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
+        }
+        outPoint.x = this.x - other.x;
+        outPoint.y = this.y - other.y;
+
+        return outPoint;
+    },
+
+    /**
+     * Multiplies component-wise `other` and `this` points and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method multiply
+     * @memberof PIXI.Point#
+     * @param {IPointData} other - The point to multiply with `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the component-wise multiplication.
+     */
+    /**
+     * Multiplies component-wise `other` and `this` points and outputs into `outPoint` or a new Point.
+     *
+     * _Note: Only available with **@pixi/math-extras**._
+     *
+     * @method multiply
+     * @memberof PIXI.ObservablePoint#
+     * @param {IPointData} other - The point to multiply with `this`.
+     * @param {IPointData} [outPoint] - A Point-like object in which to store the value,
+     * optional (otherwise will create a new Point).
+     * @returns {IPointData} The `outPoint` reference or a new Point, with the component-wise multiplication.
+     */
+    multiply<T extends IPointData>(other: IPointData, outPoint?: T): T
+    {
+        if (!outPoint)
+        {
+            (outPoint as any) = new Point();
```

Comment:
This behavior of creating new objects for the destination deviates from the current behavior. Is there any specific reason why?

Reply:
This allows for in-place operation if needed. Similar to the `toLocal` and `toGlobal` behaviour.
If the user needs every single drop of ram and performance, we use their provided object. If they don't care we create a new one.